Bookworm network man

nikeflight opened the pull request:
1. Add piaware_config.py to read from piaware config folders.
2. Add script to generate NetworkManager file
3. Edit piaware-support.install to point to new bookworm files.

nikeflight committed:
added systemd files and added bookworm generate file

nikeflight committed:
adding metadataconfig

nikeflight committed:
fix config issues

nikeflight committed:
added more piaware_config

nikeflight committed:
change bookworm-config

nikeflight committed:
delete bookworm file

nikeflight committed:
completed config read func

nikeflight committed:
add fixes for nmconnection file

nikeflight committed:
add restart network, bookworm to prepare-build.sh, add sync to generate bookworm config

Commit:
debian bookworm

nikeflight committed:
removed dch

nikeflight committed:
Merge branch 'bookworm-network-man' of https://github.com/flightaware/piaware-support into bookworm-network-man

eric1tran committed:
Clean up debian-bookworm directory

eric1tran committed:
Remove git config files

nikeflight committed:
wait 10s to let NM reboot before re-upping

nikeflight committed:
Merge branch 'bookworm-network-man' of https://github.com/flightaware/piaware-support into bookworm-network-man

nikeflight committed:
copy NetworkManager files over

nikeflight committed:
add systemd service

nikeflight committed:
add restart bookworm script to bookwomr install

nikeflight committed:
add empty body to bookworm in prepare-build

eric1tran commented on scripts/piaware_config.py:
```diff
@@ -0,0 +1,388 @@
+
+PIAWARE_CONFIG_ENUMS["receiver"] = ["rtlsdr", "sdr", "bladerf", "beast", "relay", "radarcape", "radarcape-local", "other", "none"]
+PIAWARE_CONFIG_ENUMS["uat_receiver"] = ["sdr", "stratuxv3", "other", "none"]
+PIAWARE_CONFIG_ENUMS["network_config_style"] = ["sdr", "stratuxv3", "other", "none"]
```
copy paste error? The values for these should be the [Debian OS names](https://github.com/flightaware/piaware/blob/master/package/fa_piaware_config.tcl#L61). 

[generate-network-config](https://github.com/flightaware/piaware-support/blob/master/scripts/generate-network-config#L418) uses this setting to determine how to generate the config. Is this setting now obsolete in Bookworm since we have a separate generate-network-config-bookworm?

eric1tran commented on scripts/piaware_config.py:
```diff
@@ -0,0 +1,388 @@
+
+    
+
+    def get(self, setting_key: str) -> any:
```
Is there support for setting a config?

nikeflight replied:
There is in the original TCL config.

I remember Oliver and I had decided that we not implement setting for now. 
Since we're doing a large re-write and these values don't change often.

eric1tran reviewed (changes requested):
Can you add some file descriptions / comments to the new python scripts? I think an explanation that it's a rewrite of https://github.com/flightaware/piaware/blob/master/package/fa_piaware_config.tcl as a python package and how it works would be useful

nikeflight committed:
addressing PR comments

nikeflight committed:
started unittest

nikeflight committed:
finished unit tests

nikeflight committed:
Modify  debian-bookworm/piaware-support.generate-network-config.service to use python3 to call generate-network-config-bookworm.py
Add and Modify Rules to debian-bookworm to use generate-network-config-bookworm.service
Modify piaware_config.py to add back in default network_config_style since that option is still in piaware-image-config.txt

nikeflight committed:
merge in debian-bookworm/control from dev

eric1tran commented on scripts/piaware_config.py:
```diff
@@ -0,0 +1,402 @@
+        self.settings["image-type"] = MetadataSettings(setting_type="str")
+        self.settings["manage-config"] = MetadataSettings(setting_type="bool", default=False)
+        self.settings["feeder-id"] = MetadataSettings(setting_type="UUID")
+        self.settings["force-mac-address"] = MetadataSettings(setting_type="MAC")
```
This should be `force-macaddress`

eric1tran commented on scripts/piaware_config.py:
```diff
@@ -0,0 +1,402 @@
+        self.settings["wireless-broadcast"] = MetadataSettings(sdonly=True, network=True)
+        self.settings["wireless-netmask"] = MetadataSettings(sdonly=True, network=True)
+        self.settings["wireless-gateway"] = MetadataSettings(sdonly=True, network=True)
+        self.settings["wireless-nameserver"] = MetadataSettings(default = ["8.8.8.8", "8.8.4.4"], sdonly=True, network=True)
```
wireless-nameservers

eric1tran commented on tests/test_piaware_config.py:
```diff
@@ -0,0 +1,211 @@
+        assert testm.convert_value("manage-config", "no") == False
+        assert testm.convert_value("priority", "1") == 1
+        assert testm.convert_value("adaptive-min-gain", "1.2") == 1.2
+        assert testm.convert_value("force-mac-address", "01:23:45:67:89:AB") == "01:23:45:67:89:AB"
```
force-macaddress

eric1tran commented on package/helpers/piaware-restart-network-bookworm:
Should add this file to package/helpers/Makefile

nikeflight committed:
PR comments
1. Add piaware-restart-network-bookworm to package/helpers/Makefile
2. change force-mac-address to force-macaddress
3. wireless-nameserver to wireless-nameservers

nikeflight committed:
fix type in makefike

nikeflight committed:
Merge branch 'dev' into bookworm-network-man

nikeflight committed:
skyaware version
